jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -2365,7 +2372,32 @@ public T newInstance() throws IllegalAccessException, InstantiationException {
 		checkNonSunProxyMemberAccess(security, callerClassLoader, Member.PUBLIC);
 	}
 
+/*[IF JAVA_SPEC_VERSION >= 12]*/
+	Class<?> callerClazz = getStackClass(1);
```

Comment:
This call is expensive and might affect performance. I wonder if a less expensive check can be used instead, such as checking `jdk.internal.misc.VM.isBooted()`.

Reply:
What about `Reflection.getCallerClass()` ?

Reply:
`Reflection.getCallerClass()` will have the same cost, it does the same thing.